File: 8_kyu/Compare_within_margin.cs
public class Kata
{
  public static int CloseCompare(double a, double b, double margin = 0)
  {
    double delta = a - b;
    if (delta < -margin)
      return -1;
    else if (delta > margin)
      return +1;
    else
      return 0;
  }
}

File: 8_kyu/Compare_within_margin.go
package kata

// Compare with margins.
// a == b ± margin:  0
// a  < b - margin: -1
// a  > b + margin: +1
func CloseCompare(a, b, margin float64) int {
	delta := a - b
	switch {
	case delta < -margin:
		return -1
	case delta > margin:
		return 1
	default:
		return 0
	}
}
